apps/aecore/test/aecore_suite_utils.erl

```diff
@@ -0,0 +1,359 @@
+    config_apply_options(Node, Cfg1, T);
+config_apply_options(Node, Cfg, [{add_peers, true}| T]) ->
+    Cfg1 = Cfg#{<<"peers">> =>
+              [peer_uri(N1) || N1 <- [dev1, dev2, dev3] -- [Node]]},
+    config_apply_options(Node, Cfg1, T).
+
+write_config(F, Config) ->
+    JSON = jsx:prettify(jsx:encode(Config)),
+    {ok, Fd} = file:open(F, [write]),
+    ct:log("Writing config (~p)~n~s", [F, JSON]),
+    try io:fwrite(Fd, "~s~n", [JSON])
+    after
+        file:close(Fd)
+    end,
+    VRes = aeu_env:check_config(F),
+    ct:log("Config (~p) check: ~p", [F, VRes]),
+    {ok,_} = VRes.
+
+default_config(N, Config) ->
+    {A,B,C} = os:timestamp(),
+     #{<<"keys">> =>
+           #{<<"dir">> => iolist_to_binary(keys_dir(N, Config)),
+             <<"password">> => iolist_to_binary(io_lib:format("~w.~w.~w", [A,B,C]))},
+       <<"logging">> =>
+           #{<<"hwm">> => 500},
+       <<"mining">> =>
+           #{<<"autostart">> => false},
+       <<"chain">> =>
+           #{<<"persist">> => true},
+       <<"websocket">> =>
+           #{<<"internal">> =>
+                 #{<<"acceptors">> => 10}
+            }
+      }.
+
+epoch_config_dir(N, Config) ->
+    filename:join(data_dir(N, Config), "epoch.json").
+
+%% dirs 
+node_shortcut(N, Config) ->
+    filename:join(shortcut_dir(Config), N).
+
+shortcut_dir(Config) ->
+    Top = ?config(top_dir, Config),
+    SymlinkName = ?config(symlink_name, Config),
+    filename:join([Top, "_build/test/logs", SymlinkName]).
+
+priv_dir(Config) ->
+    SubDir = ?config(test_dir, Config), 
+    filename:join(?config(priv_dir, Config), SubDir).
+
+data_dir(N, Config) ->
+    filename:join(node_shortcut(N, Config), "data").
+
+keys_dir(N, Config) ->
+    filename:join(data_dir(N, Config), "keys").
+
+delete_file(F) ->
+    case file:delete(F) of
+        ok -> ok;
+        {error, enoent} -> ok;
+        Other ->
+            erlang:error(Other, [F])
+    end.
+
+%% ============================================================
+%% Proxy process
+%% ============================================================
+
+-define(PROXY, epoch_multi_node_test_proxy).
+
+
+-define(PROXY_CALL_RETRIES, 5).
+
+call_proxy(N, Req) ->
+    call_proxy(N, Req, ?PROXY_CALL_RETRIES, 3000).
+
+call_proxy(N, Req, Tries, Timeout) when Tries > 0 ->
+    Ref = erlang:monitor(process, {?PROXY, N}),
+    {?PROXY, N} ! {self(), Ref, Req},
+    receive
+        {'DOWN', Ref, _, _, noproc} ->
+            ct:log("proxy not yet there, retrying in 1 sec...", []),
+            receive
+            after 1000 ->
+                    call_proxy(N, Req, Tries-1, Timeout)
+            end;
+        {'DOWN', Ref, _, _, Reason} ->
+            error({proxy_died, N, Reason});
+        {Ref, Result} ->
+            erlang:demonitor(Ref),
+            Result
+    after Timeout ->
+            error(proxy_call_timeout)
+    end;
+call_proxy(N, _, _, _) ->
+    erlang:error({proxy_not_running, N}).
+
+
+hostname() ->
```

Comment:
Perhaps the this, and the functions below, should be moved up, so that they don't appear to be part of the proxy_process implementation?